x/concentrated-liquidity/math/math.go

```diff
@@ -40,12 +42,14 @@ func Liquidity1(amount osmomath.Int, sqrtPriceA, sqrtPriceB osmomath.BigDec) osm
 
 	// We convert to BigDec to avoid precision loss when calculating liquidity. Without doing this,
 	// our liquidity calculations will be off from our theoretical calculations within our tests.
+	// TODO (perf): consider better conversion helpers to minimize reallocations.
 	amountBigDec := osmomath.BigDecFromDec(amount.ToLegacyDec())
 	diff := sqrtPriceB.Sub(sqrtPriceA)
 	if diff.IsZero() {
 		panic(fmt.Sprintf("liquidity1 diff is zero: sqrtPriceA %s sqrtPriceB %s", sqrtPriceA, sqrtPriceB))
 	}
 
+	// TODO (perf): consider Dec() function that does not reallocate
 	return amountBigDec.QuoMut(diff).Dec()
```

Comment:
Note: this should be addressed by https://github.com/osmosis-labs/osmosis/pull/6261, CC: @pysel